### ISSUE TITLE
if you switch team before the hit, kill seems to be count as teamkill

### ISSUE THREAD

**Issue by illwieckz** [Unvanquished/Unvanquished#845]
The infinite fire bug  (see #844) revealed some other bugs, like this one.

![friendly fire!](http://dl.illwieckz.net/b/unvanquished/bugs/infinite-fire/unvanquished_2015-11-20_034429_000.jpg)
![friendly fire!](http://dl.illwieckz.net/b/unvanquished/bugs/infinite-fire/unvanquished_2015-11-20_034354_000.jpg)

For example, that fire was "binded" to me when I was human, so, after a team switch, alien death by fire are count as team kill since the fire was _“a fire from illwieckz when he was human; but illwieckz is now alien, so it's a fire from an alien teammate”_ (oh yes, that's probably why we call it "friendly fire").

So, it looks like if you switch team, kills happened _after_ the switch but due to something you did _before_ you switch is count as teamkill.

This bug must be reproduced without that fire bug (unless the code shows it's true), because in fact I already don't know why this fire was "binded" to me as human, first.

**Comment by Viech** [comment]
Yeah, the client handling in the gamelogic is just bad. There are three different fields responsible for team-membership and even I don't know how they differ (and I broke the game badly when I tried to refactor it)…

This is a bug by design, I'll mark this issue as a cleanup task (in the sense of a CBSE-migration of client handling.)

**Comment by illwieckz** [comment]
I just faced a similar behavior: I built an forward base as human then I switched team and destroyed it as alien. Once I destroyed it I got a message saying _"a forward base is under attack"_.

Oh, and it looks like there is another bug since eternity: the message "a forward base in under attack" looks to be printed when the forward base is completely destroy, meaning it's too late to come to protect it. Am I the only one who experienced that?

**Comment by Viech** [comment]
@illwieckz, it's fine if "a forward base is under attack" is issued even when it's friendly fire. The message *should* be issued when the first structure of a forward dies, as to give the attacker a small time advantage. At some point previously, it was coupled to whether there was a repeater or stuff like that, but I feel just giving the warning on the first dead buildable to both teams is a simpler and better approach.

**Comment by illwieckz** [comment]
What I want to say is I destroyed a _human base_ as _alien_, so it's definitely not my forward base and it's a bug if I receive alerts for the opponent team. I was completely alone on the server without bots, I was training myself to attack rocket pod as mantis. I built some rocket pods as human then switched team then, being a mantis, attacked the rocket pods, then received as mantis the alert when the last rocket pod died. The rocket pod could'nt be mine since I was a mantis. Yes I built the rocket pod, but I switched team to become an alien to destroy it.

> I feel just giving the warning on the first dead buildable

Ah, If I remember I got it on the last. Well, I'm sure I got it on the last, but I don't remember if I haven't got it before while was destroying the first ones. I will do some tests.
